pandas/core/generic.py

```diff
@@ -196,7 +196,11 @@ def __unicode__(self):
     def _dir_additions(self):
         """ add the string-like attributes from the info_axis """
         additions = set([c for c in self._info_axis
-                         if isinstance(c, string_types) and isidentifier(c)])
+                         if isinstance(c, string_types) and isidentifier(c)] +
+                        [c[0] for c in self._info_axis
```

Comment:
I would rather both of these to (private) methods on ``Index`` and overriden on a ``MultiIndex``; this become much simpler

e.g.
```
def _to_identifiers(self):
    return [c for c in self if isinstance(c, string_types) and isidentifier(c)]
```

with some doc-strings & tests

then this method becomes really simple.

Reply:
Now that I understand the code better, I realize that another simple implementation could be:
```python
additions = set([c for c in self._info_axis.get_level_values(0)
                 if isinstance(c, string_types) and isidentifier(c)])
```
Please tell me which method would be best.

Reply:
yes this looks reasonable. pls add a comment on what is going one.